diff -ruN '--exclude=*git*' '--exclude=*dtc*' '--exclude=*.orig' '--exclude=*.rej' ../ubuntu-focal.orig/debian/rules ./debian/rules
--- ../ubuntu-focal.orig/debian/rules	2020-03-09 20:01:27.000000000 +1100
+++ ./debian/rules	2020-03-10 20:06:58.424886566 +1100
@@ -40,14 +40,14 @@
 do_tools_perf_jvmti?=false
 
 # Don't build tools or udebs in a cross compile environment.
-ifneq ($(DEB_HOST_ARCH),$(DEB_BUILD_ARCH))
-	do_tools=false
-	disable_d_i=true
-	do_zfs=false
-	do_dkms_nvidia=false
-	do_dkms_vbox=false
-	do_dkms_wireguard=false
-endif
+#ifneq ($(DEB_HOST_ARCH),$(DEB_BUILD_ARCH))
+#	do_tools=false
+#	disable_d_i=true
+#	do_zfs=false
+#	do_dkms_nvidia=false
+#	do_dkms_vbox=false
+#	do_dkms_wireguard=false
+#endif
 
 # Are any of the kernel signing options enabled.
 any_signed=$(sort $(filter-out false,$(uefi_signed) $(opal_signed) $(sipl_signed)))
