diff -ruN '--exclude=*git*' ../ubuntu-eoan.orig/debian.master/config/amd64/config.common.amd64 ./debian.master/config/amd64/config.common.amd64
--- ../ubuntu-eoan.orig/debian.master/config/amd64/config.common.amd64	2019-10-30 11:17:16.000000000 +1100
+++ ./debian.master/config/amd64/config.common.amd64	2019-10-30 22:40:55.656546238 +1100
@@ -18,6 +18,7 @@
 CONFIG_ANDROID=y
 CONFIG_APDS9802ALS=m
 CONFIG_APPLICOM=m
+CONFIG_APPLE_IBRIDGE_ALS=y
 CONFIG_ARCH_DEFCONFIG="arch/x86/configs/x86_64_defconfig"
 CONFIG_ARCH_MMAP_RND_BITS=28
 CONFIG_ARCH_MMAP_RND_BITS_MAX=32
@@ -183,6 +184,7 @@
 CONFIG_HFS_FS=m
 CONFIG_HIBERNATION=y
 CONFIG_HID=m
+CONFIG_HID_APPLE_IBRIDGE_TB=y
 CONFIG_HIO=m
 CONFIG_HMC6352=m
 CONFIG_HOTPLUG_PCI=y
@@ -311,6 +313,7 @@
 CONFIG_MEM_SOFT_DIRTY=y
 CONFIG_MFD_88PM800=m
 CONFIG_MFD_88PM805=m
+CONFIG_MFD_APPLE_IBRIDGE=y
 CONFIG_MFD_ARIZONA_I2C=m
 CONFIG_MFD_AXP20X_I2C=m
 CONFIG_MFD_BCM590XX=m
