diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian/dkms-versions ./debian/dkms-versions
--- ../ubuntu-disco.orig/debian/dkms-versions	2019-06-18 17:34:19.000000000 +1000
+++ ./debian/dkms-versions	2019-08-14 17:45:18.718392547 +1000
@@ -1,3 +1,2 @@
-spl-linux 0.7.12-1ubuntu3
 zfs-linux 0.7.12-1ubuntu5
 nvidia-graphics-drivers-390 390.116-0ubuntu1
diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian/rules.d/2-binary-arch.mk ./debian/rules.d/2-binary-arch.mk
--- ../ubuntu-disco.orig/debian/rules.d/2-binary-arch.mk	2019-06-18 17:34:19.000000000 +1000
+++ ./debian/rules.d/2-binary-arch.mk	2019-08-14 17:45:49.090314558 +1000
@@ -396,7 +396,6 @@
 	install -d $(dkms_dir) $(dkms_dir)/headers $(dkms_dir)/build $(dkms_dir)/source
 	cp -rp "$(hdrdir)" "$(indep_hdrdir)" "$(dkms_dir)/headers"

-	$(if $(filter true,$(enable_zfs)),$(call build_dkms, $(mods_pkg_name)-$*, $(pkgdir)/lib/modules/$(abi_release)-$*/kernel, spl, pool/universe/s/spl-linux/spl-dkms_$(dkms_spl_linux_version)_all.deb))
 	$(if $(filter true,$(enable_zfs)),$(call build_dkms, $(mods_pkg_name)-$*, $(pkgdir)/lib/modules/$(abi_release)-$*/kernel, zfs, pool/universe/z/zfs-linux/zfs-dkms_$(dkms_zfs_linux_version)_all.deb))

 ifeq ($(do_dkms_nvidia),true)
diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/amd64/generic.modules ./debian.master/abi/{prev_abi_version}/amd64/generic.modules
--- ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/amd64/generic.modules	2019-06-18 17:34:19.000000000 +1000
+++ ./debian.master/abi/{prev_abi_version}/amd64/generic.modules	2019-08-14 17:47:08.114112819 +1000
@@ -4479,7 +4479,6 @@
 spidev
 spinand
 spl
-splat
 spmi
 sr9700
 sr9800
@@ -5331,12 +5330,12 @@
 zl10039
 zl10353
 zl6100
+zlua
 znvpair
 zopt2201
 zpa2326
 zpa2326_i2c
 zpa2326_spi
-zpios
 zr36016
 zr36050
 zr36060
diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/arm64/generic.modules ./debian.master/abi/{prev_abi_version}/arm64/generic.modules
--- ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/arm64/generic.modules	2019-06-18 17:34:19.000000000 +1000
+++ ./debian.master/abi/{prev_abi_version}/arm64/generic.modules	2019-08-14 17:47:52.026001384 +1000
@@ -4668,7 +4668,6 @@
 spidev
 spinand
 spl
-splat
 spmi
 spmi-pmic-arb
 sprd-dma
@@ -5554,12 +5553,12 @@
 zl10039
 zl10353
 zl6100
+zlua
 znvpair
 zopt2201
 zpa2326
 zpa2326_i2c
 zpa2326_spi
-zpios
 zr364xx
 zram
 zstd
diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/ppc64el/generic.modules ./debian.master/abi/{prev_abi_version}/ppc64el/generic.modules
--- ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/ppc64el/generic.modules	2019-06-18 17:34:19.000000000 +1000
+++ ./debian.master/abi/{prev_abi_version}/ppc64el/generic.modules	2019-08-14 17:49:45.713714774 +1000
@@ -4164,7 +4164,6 @@
 spidev
 spinand
 spl
-splat
 spmi
 sr9700
 sr9800
@@ -4952,12 +4951,12 @@
 zl10039
 zl10353
 zl6100
+zlua
 znvpair
 zopt2201
 zpa2326
 zpa2326_i2c
 zpa2326_spi
-zpios
 zr364xx
 zram
 zstd
diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/s390x/generic.modules ./debian.master/abi/{prev_abi_version}/s390x/generic.modules
--- ../ubuntu-disco.orig/debian.master/abi/{prev_abi_version}/s390x/generic.modules	2019-06-18 17:34:19.000000000 +1000
+++ ./debian.master/abi/{prev_abi_version}/s390x/generic.modules	2019-08-14 17:49:02.809822637 +1000
@@ -745,7 +745,6 @@
 smsgiucv_app
 softdog
 spl
-splat
 st
 st_drv
 ste10Xp
@@ -955,9 +954,9 @@
 zcrypt_cex4
 zfcp
 zfs
+zlua
 zlib_deflate
 znvpair
-zpios
 zram
 zstd
 zstd_compress
