diff -ruN '--exclude=*git*' ../ubuntu-disco.orig/debian.master/config/config.common.ubuntu ./debian.master/config/config.common.ubuntu
--- ../ubuntu-disco.orig/debian.master/config/config.common.ubuntu	2019-06-18 17:34:19.000000000 +1000
+++ ./debian.master/config/config.common.ubuntu	2019-07-07 18:55:57.754837264 +1000
@@ -5564,7 +5564,7 @@
 # CONFIG_MODULE_FORCE_UNLOAD is not set
 CONFIG_MODULE_SIG=y
 CONFIG_MODULE_SIG_ALL=y
-# CONFIG_MODULE_SIG_FORCE is not set
+CONFIG_MODULE_SIG_FORCE=y
 CONFIG_MODULE_SIG_HASH="sha512"
 CONFIG_MODULE_SIG_KEY="certs/signing_key.pem"
 # CONFIG_MODULE_SIG_SHA1 is not set
